RunTests.ps1

```
@@ -67,7 +67,26 @@ Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1"
 try 
 {
     #region Prepare / Clean the powershell console.
-    $WorkingDirectory = Split-Path -parent $MyInvocation.MyCommand.Definition 
+    $MaxDirLength = 32
+    $WorkingDirectory = Split-Path -parent $MyInvocation.MyCommand.Definition
+    Set-Variable -Name shortRandomNumber -Value $(Get-Random -Maximum 99999 -Minimum 11111) -Scope Global
+    Set-Variable -Name shortRandomWord -Value $(-join ((65..90) | Get-Random -Count 4 | ForEach-Object {[char]$_})) -Scope Global
+    if ( $WorkingDirectory.Length -gt $MaxDirLength)
+    {
+        $OriginalWorkingDirectory = $WorkingDirectory
+        Write-Host "Current working directory '$WorkingDirectory' length is greather than $MaxDirLength. Need to change the working directory."
+        $TempWorkspace = "$(Split-Path $OriginalWorkingDirectory -Qualifier)"
+        New-Item -ItemType Directory -Path "$TempWorkspace\LISAv2" -Force -ErrorAction SilentlyContinue | Out-Null
+        New-Item -ItemType Directory -Path "$TempWorkspace\LISAv2\$shortRandomWord$shortRandomNumber" -Force -ErrorAction SilentlyContinue | Out-Null
+        $finalWorkingDirectory = "$TempWorkspace\LISAv2\$shortRandomWord$shortRandomNumber"
+        $tmpSource = '\\?\' + "$OriginalWorkingDirectory\*"
+        Write-Host "Copying current workspace to $finalWorkingDirectory"
+        Copy-Item -Path $tmpSource -Destination $finalWorkingDirectory -Recurse -Force -ErrorAction SilentlyContinue| Out-Null
```

this is a vital operation and the erroraction should not be set to silently continue